elasticai/creator/ir/graph_delegate_test.py

```diff
@@ -0,0 +1,69 @@
+from .graph_delegate import GraphDelegate
+from .graph_iterators import bfs_iter_up, dfs_pre_order
+
+
+def test_iterating_breadth_first_upwards():
+    g = GraphDelegate()
+    """
+             0
+             |
+          /-----\
+          |     |
+          1     2
+          | /---+
+          |/    |
+          3     4
+          |     |
+          |     6
+          |/----+
+          5
+    """
+    g = GraphDelegate.from_dict(
+        {
+            "0": ["1", "2"],
+            "1": ["3"],
+            "2": ["3", "4"],
+            "3": ["5"],
+            "4": ["6"],
+            "6": ["5"],
+        }
+    )
+
+    actual = tuple(bfs_iter_up(g.get_predecessors, "5"))
+    assert set(actual[0:2]) == {"3", "6"}
+    assert (set(actual[2:4]) == {"1", "2"} and actual[4] == "4") or (
+        set(actual[3:5]) == {"1", "2"} and actual[2] == "4"
```

Comment:
Does this actually vary or did you just show that both is correct?

Reply:
annoyingly it can vary based on order of insertion and i didn't want to think about what order would be correct in this case :(.

Reply:
For the least confusion, if it just depends on the input dict, I would find out which one is first and just assert that. If you're too lazy maybe a comment that the result is always the same when the input is the same.

Reply:
nah.. will do

Reply:
and that change even shows a bug now... node `"0"` being yielded twice in the end ><